### PR TITLE
Add Elasticache Create Tests

### PR DESCRIPTION
**Pull request by tedivm**
Adds the tests for creating a redis cluster.

### CODE REVIEW

File: test/basic/elasticache/elasticache_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/aws/aws-sdk-go-v2/service/elasticache"
+	"github.com/aws/aws-sdk-go-v2/service/elasticache/types"
+
+	//"github.com/aws/aws-sdk-go/aws/awserr"
+	//"github.com/aws/aws-sdk-go/aws/session"
+	legacy_aws_sdk_ec2 "github.com/aws/aws-sdk-go/service/ec2"
+
+	terratest_aws "github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
```
Comment (madhuravius) [inline test/basic/elasticache/elasticache_test.go:20]: we're gonna add testify?

**Review by madhuravius — COMMENTED** [review]
LGTM - i'd just drop all the commented blocks for now i think (?)

File: test/basic/elasticache/elasticache_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		redisId,
+	)
+	assert.Nil(t, redisErr)
+	assert.Equal(t, redisAsset.Id, redisId)
+	assert.Equal(t, redisAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+	assert.NotNil(t, redisAsset.Outputs)
+
+	redis_cluster_id_base := redisAsset.GetOutputs()["elasticache_cluster_id"].Data.(string)
+
+	cluster_details, err := GetCluster(redis_cluster_id_base + "-001")
```
Comment (madhuravius) [inline test/basic/elasticache/elasticache_test.go:104]: i think our cluster id output is probably wrong from the terraform and we're sending up a different id

```
output "elasticache_cluster_id" {
  value = module.elasticache_redis.aws_elasticache_cluster.id
}
```

maybe we want a different id here? `aws_elasticache_replication_group` should be output on the asset bundle instead maybe? or something else? i think is what we might actually want: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group#member_clusters

and then query on these

Comment (madhuravius) [reply test/basic/elasticache/elasticache_test.go:100]: follow up ticket made, resolving

**Review by madhuravius — APPROVED** [review]
LGTM

**Review by aguilinger — APPROVED** [review]

**Pull request merged**